### PR TITLE
Blog App - Controllers

### PR DESCRIPTION
**Pull request by anyars-encarta**
In this project, I implemented functionalities to meet the following requirements:

- [ ] The app handles the following URLs:

  - https://
  - https://users/745
  - https://users/745/posts
  - https://users/745/posts/3

- [ ] For each URL you I created:

  - A route.
  - An action in the correct controller.
  - A view file.

- [ ] View files include only placeholders written in plain HTML.

- [ ] I added code to handle only the 4 required URLs.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+  def show
+    # Action to handle https://users/745
+    @user = User.find(params[:user_id])
+  end
+end
```
Comment (Munsa1) [inline app/controllers/users_controller.rb]: - [ ] Kindly add the with #index action to your user controller as per requirement. t's okay for the action to be empty for now.

e.g
`def index; end`

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<!-- Placeholder for user details -->
+<h1>User Details</h1>
+<p>This is where user details will be displayed.</p>
```
Comment (Munsa1) [inline app/views/users/show.html.erb:3]: - [ ] Kindly add a view `index.html.erb` corresponding to the `index` action created in the user controller

File: config/routes.rb [hunk 3]
```diff
@@ -6,5 +6,11 @@
...
   # Defines the root path route ("/")
-  # root "posts#index"
+  # Set root to home#index for the base URL
+  root 'home#index'
+
+  # Individual routes for users, their posts, and specific posts
+  get 'users/:user_id', to: 'users#show', as: 'user'
+  get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
 end
```
Comment (Munsa1) [inline config/routes.rb]: - [ ] Ensure you update your routes to show the `users#index` route corresponding to the controller action and the view file as per the requirement

image

Comment (Munsa1) [inline config/routes.rb]: - [ ] Please ensure that you do not expose unnnecessary endpoints by using the `only` keyword for your routes

e.g
```



# config/routes.rb
  root 'home#index
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @anyars-encarta, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Added Actions To Controllers :heavy_check_mark: 
Code Documentation :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @anyars-encarta 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Controllers,  Views and Routes setup correctly 👏

  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**